zio/zstio/reader.go

```diff
@@ -0,0 +1,25 @@
+package zstio
+
+import (
+	"errors"
+	"io"
+
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zst"
+)
+
+type Reader struct {
+	zst.Reader
+}
+
+func NewReader(r io.Reader, zctx *resolver.Context) (*Reader, error) {
+	seeker, ok := r.(zst.Seeker)
+	if !ok {
+		return nil, errors.New("zst must be used with a seekable input")
+	}
```

Comment:
Nit: This feels wrong. Why not make the first parameter a zst.Seeker and leave the check to the caller?

Reply:
I had it the other way initially and it felt wrong that way :-). There are a few places that create a zstio.NewReader and lifting the code out to those places seemed wrong.  This zst.NewReader takes a Seeker, but it seems like this was ok to do in zstio.NewReader.  I think to get this right, we need to refactor detector a bit.  Maybe fix in a subsequent PR?